### PR TITLE
[Chef-17] 7 of X - Updating Berkshelf version

### PR DESCRIPTION
**Pull request by johnmccrae**
## Description
The version of Berkshelf on the main branch causes chef-17 to pull in a version that targets Chef-18 and causes build/test failures. I created a branch that checks which version of Ruby is loaded and pulls in the correct version accordingly.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: omnibus/Gemfile [hunk 1]
```diff
@@ -6,14 +6,15 @@ gem "artifactory"
 
 gem "pedump"
 
+# Use Berkshelf for resolving cookbook dependencies
+# Using a branch here that calls out a berkshelf version for chef-17
+gem "berkshelf", git: "https://github.com/chef/berkshelf.git", branch: "jfm/chef17_berkshelf_tweak"
```
Comment (ramereth) [inline omnibus/Gemfile:11]: Will this branch not be released?

Comment (johnmccrae) [reply omnibus/Gemfile:11]: I am guessing it might not since the Chef/Berkshelf repo is a fork from Berkshelf/Berkshelf

Comment (neha-p6) [reply omnibus/Gemfile:11]: What issue do we currently have with using `gem "berkshelf", ">= 7.0"` ?

**Review by neha-p6 — CHANGES_REQUESTED** [review]

File: omnibus/Gemfile (hunk 1 above)
Comment (johnmccrae) [reply omnibus/Gemfile:11]: It was always pulling in Chef-18 and causing build/test failures. I needed a way to get a chef-17 dependency in the mix too.

Comment (johnmccrae) [reply omnibus/Gemfile:11]: @ramereth I was wrong. I COULD merge that in. Now I just need to redo the PR.

**Comment by johnmccrae** [comment]
Trashed this branch. Closing this and starting a new branch

**Pull request closed without merging**